src/NUnitTestAdapter/NUnitEngine/EngineAdapterClasses.plantuml

```diff
@@ -0,0 +1,59 @@
+@startuml
+title NUnitEngineAdapterClasses for discovery
+   CanHaveFixtures <|-- TestSuite
+   SuiteBase <|-- CanHaveTestCases
+   SuiteBase <|-- CanHaveFixtures
+   SuiteBase <|-- TestRun
+   CanHaveTestCases <|-- Fixture
+   SuiteBase <|-- TestCase
+   CanHaveTestCases <|-- ParametrizedMethod
```

Comment:
FIxed all places except for the adaptersettings, which has been there since 3.16.1.  CHanging it there would mean a breaking change for those using that setting......